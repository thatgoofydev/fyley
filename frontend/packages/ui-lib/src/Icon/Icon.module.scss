@import "src/base-styling/vars";

.icon {
  display: inline-block;

  svg {
    fill: $icon-color
  }
}