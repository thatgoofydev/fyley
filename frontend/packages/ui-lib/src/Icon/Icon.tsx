import React, { FC } from "react";
import styles from "./Icon.module.scss";

export type IconType = "add" | "warning";

export interface IProps {
  type: IconType;
  size?: number;
}

export const Icon: FC<IProps> = ({ type, size = 24 }) => {
  const renderSwitch = () => {
    switch (type) {
      case "add":
        return (
          <path d="M12 2a10 10 0 110 20 10 10 0 010-20zm0-2a12 12 0 100 24 12 12 0 000-24zm6 13h-5v5h-2v-5H6v-2h5V6h2v5h5v2z" />
        );
      case "warning":
        return (
          <path d="M12 5l9 16H3l9-16zm0-4L0 23h24L12 1zm-1 9h2v6h-2v-6zm1 10a1 1 0 110-3 1 1 0 010 3z" />
        );
    }
  };

  return (
    <div className={styles.icon} style={{ width: `${size}px`, height: `${size}px` }}>
      <svg xmlns="http://www.w3.org/2000/svg" width={size} height={size} viewBox={`0 0 24 24`}>
        {renderSwitch()}
      </svg>
    </div>
  );
};
