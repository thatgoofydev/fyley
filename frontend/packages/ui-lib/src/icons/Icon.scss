@import "src/base/colors";

.icon {
  display: inline-block;

  svg {
    fill: $icon-color
  }
}