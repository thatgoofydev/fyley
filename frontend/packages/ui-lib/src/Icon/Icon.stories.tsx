import React from "react";
import { Meta, Story } from "@storybook/react";
import { Icon, IProps } from "./Icon";

export default {
    title: "General/Icons",
    component: Icon
} as Meta;

export const IconTemplate: Story<IProps> = args => {
    return <Icon {...args} />;
};
IconTemplate.storyName = "Icon";
IconTemplate.args = {
    type: "add",
    size: 24
};
