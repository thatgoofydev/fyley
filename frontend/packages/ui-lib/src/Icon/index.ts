export { Icon } from "./Icon";
import { IconType as tIconType } from "./Icon";
export type IconType = tIconType;
