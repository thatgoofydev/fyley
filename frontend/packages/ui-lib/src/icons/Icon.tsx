import React, { FunctionComponent } from "react";
import "./Icon.scss";

export type IconType = "add" | "edit";

export interface IconProps {
  type: IconType;
  size?: number;
}

export const Icon: FunctionComponent<IconProps> = ({ type, size = 24 }) => {
  const renderAdd = () => (
    <path d="M12 2a10 10 0 110 20 10 10 0 010-20zm0-2a12 12 0 100 24 12 12 0 000-24zm6 13h-5v5h-2v-5H6v-2h5V6h2v5h5v2z" />
  );

  const renderEdit = () => (
    <path d="M18.363 8.464l1.433 1.431-12.67 12.669L.001 24l1.439-7.127L14.105 4.205l1.431 1.431L3.281 17.86l-.726 3.584 3.584-.723L18.363 8.464zM18.307 0l-2.815 2.817 5.691 5.692L24 5.688 18.307 0zM5.989 18.718L17.302 7.402l-.705-.707L5.284 18.009l.705.709z" />
  );

  const renderSwitch = () => {
    switch (type) {
      case "add":
        return renderAdd();
      case "edit":
        return renderEdit();
    }
  };

  return (
    <div className="icon" style={{ width: `${size}px`, height: `${size}px` }}>
      <svg
        xmlns="http://www.w3.org/2000/svg"
        width={size}
        height={size}
        viewBox={`0 0 24 24`}
      >
        {renderSwitch()}
      </svg>
    </div>
  );
};
