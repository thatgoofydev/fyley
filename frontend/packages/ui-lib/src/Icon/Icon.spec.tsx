import React from "react";
import { render, cleanup } from "@testing-library/react";

import { Icon } from "./Icon";

afterEach(cleanup);

describe("<Icon />", () => {
  test("should render icon at default of size 24", () => {
    const rendered = render(<Icon type="add" />);
    const iconWrapper = rendered.container.firstChild;
    expect(iconWrapper).toHaveStyle("width: 24px; height: 24px;");
    expect(iconWrapper).toHaveClass("icon");
  });

  test("should render icon at correct size when a custom size is provided", () => {
    const rendered = render(<Icon type="add" size={200} />);
    const iconWrapper = rendered.container.firstChild;
    expect(iconWrapper).toHaveStyle("width: 200px; height: 200px;");
    expect(iconWrapper).toHaveClass("icon");
  });
});

describe('<Icon type="add" />', () => {
  test("should render add icon", () => {
    const rendered = render(<Icon type="add" />);
    expect(rendered).toMatchSnapshot();
  });
});

describe('<Icon type="warning" />', () => {
  test("should render warning icon", () => {
    const rendered = render(<Icon type="warning" />);
    expect(rendered).toMatchSnapshot();
  });
});
