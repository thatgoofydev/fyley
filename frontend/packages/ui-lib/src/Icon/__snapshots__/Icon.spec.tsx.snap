// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon type="add" /> should render add icon 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="icon"
        style="width: 24px; height: 24px;"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2a10 10 0 110 20 10 10 0 010-20zm0-2a12 12 0 100 24 12 12 0 000-24zm6 13h-5v5h-2v-5H6v-2h5V6h2v5h5v2z"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="icon"
      style="width: 24px; height: 24px;"
    >
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2a10 10 0 110 20 10 10 0 010-20zm0-2a12 12 0 100 24 12 12 0 000-24zm6 13h-5v5h-2v-5H6v-2h5V6h2v5h5v2z"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Icon type="warning" /> should render warning icon 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="icon"
        style="width: 24px; height: 24px;"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 5l9 16H3l9-16zm0-4L0 23h24L12 1zm-1 9h2v6h-2v-6zm1 10a1 1 0 110-3 1 1 0 010 3z"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="icon"
      style="width: 24px; height: 24px;"
    >
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 5l9 16H3l9-16zm0-4L0 23h24L12 1zm-1 9h2v6h-2v-6zm1 10a1 1 0 110-3 1 1 0 010 3z"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
